src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -258,17 +288,80 @@ private void EnsureGlobbingUrlBuilder()
             }
         }
 
-        private static void BuildLinkTag(IDictionary<string, string> attributes, TagHelperContent builder)
+        private void BuildLinkTag(IDictionary<string, string> attributes, TagHelperContent builder)
         {
             builder.Append("<link ");
 
             foreach (var attribute in attributes)
             {
-                builder.Append(
-                    string.Format(CultureInfo.InvariantCulture, "{0}=\"{1}\" ", attribute.Key, attribute.Value));
+                if (attribute.Key == "href")
+                {
+                    AppendHref(builder, attribute.Key, attribute.Value);
```

Comment:
The more I look at this I see that it can be simplified even further, ignore my comments in the `AppendHref` method, it can be ignored.

This piece of the foreach can be as follows:

```C#
foreach (var attribute in attributes)
{
    var attributeValue = attribute.Value;
    if (string.Equals(attribute.Key, "href", StringComparison.OrdinalIgnoreCase) &&
        FileVersion)
    {
        attributeValue = ResolveVersionedHref(attributeValue);
    }

    builder.Append(
        string.Format(CultureInfo.InvariantCulture, "{0}=\"{1}\" ", attribute.Key, attributeValue));
}
```